test/unit/c/test_gethostname.py

```diff
@@ -45,13 +45,21 @@ def opt_ml_input_config():
         [{"current_host": "algo-1-thse"}, "algo-1-thse"],
     ],
 )
+@pytest.mark.xfail(
+    "IS_CODEBUILD_IMAGE" not in os.environ,
```

Comment:
Will "IS_CODEBUILD_IMAGE=false" also trigger this test running? Can we add that as a condition?

Reply:
Specifically, can we only run this test if lowercase(variable) == true, or something along those lines?

Reply:
Good point-- the test will run regardless of what string is assigned to `IS_CODEBUILD_IMAGE`, as long as the environment variable exists. I can add a conditional to check for the string "false".